src/main/java/com/denizenscript/depenizen/bukkit/objects/residence/ResidenceTag.java

```diff
@@ -108,44 +114,73 @@ public String toString() {
         // -->
-        if (attribute.startsWith("name")) {
-            return new ElementTag(residence.getName()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.getResidence().getName());
+        });
 
         // <--[tag]
         // @attribute <ResidenceTag.owner>
         // @returns PlayerTag
         // @plugin Depenizen, Residence
         // @description
-        // Returns the owner of the residence.
+        // Returns a PlayerTag of owner in this Residence.
         // -->
-        else if (attribute.startsWith("owner")) {
-            return new PlayerTag(residence.getOwnerUUID()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(PlayerTag.class, "owner", (attribute, object) -> {
+            return new PlayerTag(object.getResidence().getOwnerUUID());
+        });
+
+        // <--[tag]
+        // @attribute <ResidenceTag.subzones>
+        // @returns ListTag
+        // @plugin Depenizen, Residence
+        // @description
+        // Returns a ListTag(ResidenceTag) of subzones in this Residence.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "subzones", (attribute, object) -> {
+            ListTag list = new ListTag();
+            for (ClaimedResidence subzone : object.getResidence().getSubzones()) {
+                list.addObject(new ResidenceTag(subzone));
+            }
+            return list;
+        });
+
+        // <--[tag]
+        // @attribute <ResidenceTag.area>
+        // @returns CuboidTag
+        // @plugin Depenizen, Residence
+        // @description
+        // Returns a CuboidTag of this Residences area.
+        // -->
+        tagProcessor.registerTag(CuboidTag.class, "area", (attribute, object) -> {
+            CuboidArea area = object.getResidence().getMainArea();
+            return new CuboidTag(area.getLowLocation(), area.getHighLocation());
+        });
 
         // <--[tag]
         // @attribute <ResidenceTag.is_within[<location>]>
         // @returns ElementTag(Boolean)
         // @plugin Depenizen, Residence
         // @description
-        // Returns whether the specified location is within this Residence.
+        // Returns boolean whether the specified location is within this Residence.
         // -->
-        else if (attribute.startsWith("is_within") && attribute.hasParam()) {
-            LocationTag location = attribute.paramAsType(LocationTag.class);
-            return new ElementTag(residence.containsLoc(location)).getObjectAttribute(attribute.fulfill(1));
-        }
-
-        return new ElementTag(identify()).getObjectAttribute(attribute);
+        tagProcessor.registerTag(ElementTag.class, "is_within", (attribute, object) -> {
+            if (attribute.hasParam()) {
+                LocationTag location = attribute.paramAsType(LocationTag.class);
```

Comment:
Should use the `registerTag` method for tags with a param, see [this](https://github.com/DenizenScript/Denizen/blob/dev/plugin/src/main/java/com/denizenscript/denizen/objects/AreaContainmentObject.java#L220-L222) for an example